bin/portal-bridge/src/bridge/e2hs.rs

```diff
@@ -0,0 +1,439 @@
+        match Self::gossip_header_by_number(
+            block_tuple.header_with_proof.header_with_proof.clone(),
+            portal_client.clone(),
+            block_stats.clone(),
+        )
+        .await
+        {
+            Ok(_) => {
+                metrics.report_gossip_success(true, "header_by_number");
+                info!("Gossiped header by number: {header_number}");
+            }
+            Err(e) => {
+                metrics.report_gossip_success(false, "header_by_number");
+                warn!("Failed to gossip header by number: {header_number} - {e:?}");
+            }
+        }
+        metrics.stop_process_timer(timer);
+
+        // gossip body
+        let timer = metrics.start_process_timer("gossip_block_body");
+        match Self::gossip_body(
+            block_tuple.body.body.clone(),
+            header_hash,
+            portal_client.clone(),
+            block_stats.clone(),
+        )
+        .await
+        {
+            Ok(_) => {
+                metrics.report_gossip_success(true, "block_body");
+                info!("Gossiped body: {header_number}");
+            }
+            Err(e) => {
+                metrics.report_gossip_success(false, "block_body");
+                warn!("Failed to gossip body: {header_number} - {e:?}");
+            }
+        }
+        metrics.stop_process_timer(timer);
+
+        // gossip receipts
+        let timer = metrics.start_process_timer("gossip_receipts");
+        match Self::gossip_receipts(
+            block_tuple.receipts.receipts.clone(),
+            header_hash,
+            portal_client.clone(),
+            block_stats.clone(),
+        )
+        .await
+        {
+            Ok(_) => {
+                metrics.report_gossip_success(true, "receipts");
+                info!("Gossiped receipts: {header_number}");
+            }
+            Err(e) => {
+                metrics.report_gossip_success(false, "receipts");
+                warn!("Failed to gossip receipts: {header_number} - {e:?}");
+            }
+        }
+        metrics.stop_process_timer(timer);
+        Ok(())
+    }
+
+    async fn gossip_header_by_hash(
+        header_with_proof: HeaderWithProof,
+        portal_client: HttpClient,
+        block_stats: Arc<Mutex<HistoryBlockStats>>,
+    ) -> anyhow::Result<()> {
+        let header_hash = header_with_proof.header.hash_slow();
+        let hwp_by_hash_content_key = HistoryContentKey::new_block_header_by_hash(header_hash);
+        let hwp_content_value = HistoryContentValue::BlockHeaderWithProof(header_with_proof);
+        if let Err(err) = gossip_history_content(
+            portal_client,
+            hwp_by_hash_content_key.clone(),
+            hwp_content_value,
+            block_stats,
+        )
+        .await
+        {
+            error!(
+                "Failed to gossip history content key: {:?} - {:?}",
+                hwp_by_hash_content_key, err
+            );
```

Comment:
```suggestion
            error!("Failed to gossip history content key: {hwp_by_hash_content_key:?} - {err:?}");
```

We inline the variables in the error logs in most functions, maybe we should do it here too to be consistent?

Reply:
could be nice if you find a way to add this to the linting rules

Reply:
I think we have the linting rule enabled, but they are still working on fmt to cover all the edge cases in where it misses it